docs/introduction.md

```diff
@@ -5,402 +5,103 @@ This document is aimed at introducing developers to the Flow Data Provisioning S
 
-Flow is often [upgraded with breaking changes](#sporks) that require a network restart. The new network with the updated version is started from a snapshot of the previous execution state.
-The final version of the previous execution state remains available through a legacy access node that connects to a legacy [execution node](#execution-nodes), but once again this is limited to the last 300 chunks.
+Flow is often upgraded with breaking changes that require a network restart. The new network with the updated version is started from a snapshot of the previous execution state.
+The final version of the previous execution state remains available through a legacy access node that connects to a legacy execution node, but once again this is limited to the last 300 chunks.
 
-## Flow
+### Indexing Past Sporks
 
-Most current blockchains are built as a homogenous system comprised of *full nodes*.
-Every full node collects and propagates the transaction it sees on the network.
-When being able to mine the next block, or being selected as the leader to propose the next block, they will choose the transactions to include according to the consensus rules.
-This means that every node needs to be able to execute every transaction it includes in a block in order to validate them.
-When validating other nodes' blocks, every node also needs to run every transaction to validate the block's correctness.
+In order to index past sporks using Flow DPS, three elements are needed from the spork to be indexed:
 
-These existing blockchain architectures thus suffer from well documented throughput limitations. Two approaches to mitigate this are layer-2 networks and sharding.
+* Its root checkpoint
+* The protocol state database from one of its consensus nodes
+* The write-ahead log from one of its execution nodes
 
-Layer-2 solutions include networks like Bitcoin's Lightning and Ethereum's Plasma — these networks work off the main chain.
+You can then run the `flow-dps-indexer` binary, giving it access to those elements like such:
 
-Sharding is a technique where a network is broken up into several interconnected networks.
-Sharding significantly increases the complexity of the programming model by breaking ACID guarantees (Atomicity, Consistency, Isolation and Durability), increasing the cost and time for application development.
+```console
+$ ./flow-dps-indexer -a -l debug -d /var/flow/data/protocol -t /var/flow/data/execution -c /var/flow/bootstrap/root.checkpoint -i /var/flow/data/index
+```
 
-Flow was designed to provide a blockchain that can scale while preserving composability in a single blockchain state.
-This is achieved by a novel approach where work traditionally assigned to full nodes is split and assigned to specific roles, allowing pipelining.
-We recognize the following roles in the Flow architecture.
+This can take a very long time to process, and require large amounts of available storage space for the new index to be created.
```

Comment:
Creating the index database can take a very long time and requires large amounts of available storage space and memory.